### PR TITLE
Make running tests available via setup.py (setuptools)

### PR DESCRIPTION
**Pull request by bittner**
- Integrate all tests of `tests.py` as test suite in setuptools
- Add `flake8` as a test (so it won't be forgotten to run)
- Add flake8 configuration to the project
- Refactor and explain tests module (docstring)
- Add development requirements (tests + docs generation)

### CODE REVIEW

**Comment by bittner** [comment]
NOTE that `python setup.py test` execute 3 tests, whereas `python manage.py test` executes 4 tests!
The latter also runs `behave_django.testcase.BehaveDjangoTestCase.runTest` as a test. Ever since.
Unfortunately, I've seen no easy way to work around this.

`flake8` is meant to be installed manually. The `test_require` option in setuptools installs the package in the egg folder (or so), and that doesn't make `flake8` available for execution in the related test.

**Comment by bittner** [comment]
The darn non-issue #14 makes the build fail for Python 2.6 + Django 1.4 again.  :unamused:

File: tests.py [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+class BehaveDjangoTestCase(unittest.TestCase):
     def test_command_should_exit_zero_if_passing(self):
-        FNULL = open(os.devnull, 'w')
-        return_code = subprocess.call(
-            ['python', 'manage.py', 'behave', '--tags', '~@failing'],
-            stdout=FNULL,
-            stderr=subprocess.STDOUT)
-        FNULL.close()
-        self.assertEqual(return_code, 0)
+        self.assertEqual(0, run_silently(
```
Comment (mixxorz) [inline tests.py]: I think assigning the exit code to a variable is easier to read.

``` python
exit_code = run_silently(....)

self.assertEqual(0, exit_code)
```

**Comment by mixxorz** [comment]
Aside from the comment I left, I think this is good! :grinning:. Running flake8 as a test was a great idea!

**Comment by bittner** [comment]
Good. BTW, I've just looked it up (was unsure myself), it's called "exist status" ([Wikipedia](https://en.wikipedia.org/wiki/Exit_status)). I'll use that.

## BTW, is py.test style okay?

Would it be okay to use the assert syntax promoted by [py.test](http://pytest.org/latest/)? Personally, I prefer it to the common boiler-plate syntax. What do you think?

**mixedCase boiler-plate functions:** (the "prevailing style" in unit testing, see [PEP 0008](https://www.python.org/dev/peps/pep-0008/#function-names))

``` python
self.assertEqual(exit_code, 0)
```

**assert style promoted by py.test:** (usually simple output, optimized output when hit by py.test)

``` python
assert exit_status == 0
```

I've just checked, the inverted order (popularly used I think in JavaScript in recent years) is not necessary with `assert`:

``` python
assert 0 == exit_status  # maybe less readable (unnatural order), not necessary because ...
assert exit_status = 0  # triggers a SyntaxError (comparison enforced by assert)
```

## Use py.test, or is assert alone fine enough?

I may use the simple assert. I could also add py.test to the development requirements and make setup.py run py.test. The output would be nicer, but it's an additional requirement. The test code would be shorter too (no TestCase class but all functions), but not everyone knows it. What's your preference?

**Comment by bittner** [comment]
I'd love to have this PR merged into master before continuing with PR #28. Do you think that's possible?

I don't care whether we leave the tests as they are (a TestCase class) or I convert them to py.test (plain functions). The latter would mean some integration work with setuptools. We can safely skip this job.

**Comment by mixxorz** [comment]
I think sticking to unittest is fine. The only reason why I wanted `exit_status` to be in a variable is so that it's clear that we're testing of `exit_status` is 0. Haha.

Anyway. Looks good.

**Pull request merged**